src/Home.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState, FunctionComponent } from 'react';
+import './App.scss';
+import CardGrid from './CardGrid';
+import SearchFilter from './SearchFilter';
+import { PokemonAttributes, PokemonFilter } from './types/index';
+
+interface Props {
+  allPokemon: Array<PokemonAttributes>;
+  isSearchVisible: boolean;
+  handleUnmount: () => void;
+}
+
+const Home: FunctionComponent<Props> = ({
+  allPokemon,
+  isSearchVisible,
+  handleUnmount,
+}) => {
+  const [filter, setFilter] = useState<PokemonFilter>({ name: '', types: [] });
+
+  // ensure search component is unmounted when cardgrid is unmounted
+  useEffect(() => {
+    return () => handleUnmount();
+  }, [handleUnmount]);
```

Comment:
I don't think this is necessary. If the `Home` component is unmounted, the search would also be unmounted because it's in the `Home` component.

Reply:
The state of whether the `Search` component is visible or not (`isSearchVisible`) is stored in `App`, which means I need this to ensure that `isSearchVisible` is set to `false` when the `Home` component is unmounted. I've updated the comment to be more accurate.

Reply:
> The state of whether the Search component is visible or not (isSearchVisible) is stored in App

Why? If the behavior you want to is to show the search filter when we're on the home page, and not show the search filter when we're not on the home page, then get rid of `isSearchVisible`, it's not doing anything.

Maybe I'm missing something in the desired UX, let's chat about it tomorrow.

Reply:
The search button in `Nav`, which is a sibling of `Home`, toggles `Search` within `Home`. Because `Nav` is a sibling of `Home`, the state of whether `Search` is visible or not lives in `App`, their mutual parent.

Reply:
The way I have this set up right now works behaviorally, but doesn't pass muster because I'm use `useEffect` in a janky way. Could use your input on how to better design `App` and `Home` to facilitate toggling `Search`. Is it bad practice to have the search button in `Nav` be a link _and_ a toggle button? Should a toggle button be added to `Search` instead?

Reply:
Discussed in person. We won't have the search button performing double duty as navigational link and visibility toggle.